addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -397,8 +549,12 @@ private void generateHttpMetadataHeaders( final Transfer transfer, final List<Tr
             return;
         }
 
-        Response responseWithLastModified =
-                Response.fromResponse( response ).lastModified( new Date( transfer.lastModified() ) ).build();
+        Response customizedResponse = Response.fromResponse( response )
+                                              .header( ApplicationHeader.content_length.upperKey(), transfer.length() )
+                                              .header( ApplicationHeader.indy_origin.upperKey(), null )
```

Comment:
Does NPM error out when we add an Indy-Origin header??

Reply:
nope, @jdcasey, if saved here, it will appear two repeated Indy-Origin headers in the response (doHead will set the headers with exited http-meta), so abandon one from the npm customized http-meta(produced in doGet), to make sure the headers more clear. It seems npm accepts this kind of custom field, not found it was blocked by it until now.